js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,156 @@
+// Package fs provides a k6 module that allows users to interact with files from the
+// local filesystem.
+package fs
+
+import (
+	"fmt"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/lib/fsext"
+)
+
+// TODO @oleiade: we need to explicitly document that File are read-only and are really views, thus
+// any changes made to the file from outside of k6, while it runs, will be ignored as k6 will work
+// on its own cached version of the file's content.
+
+type (
+	// RootModule is the global module instance that will create instances of our
+	// module for each VU.
+	RootModule struct {
+		fileRegistry fileRegistry
+	}
+
+	// ModuleInstance represents an instance of the fs module for a single VU.
+	ModuleInstance struct {
+		vu modules.VU
+
+		fileRegistry *fileRegistry
+	}
+)
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &ModuleInstance{}
+)
+
+// New returns a pointer to a new [RootModule] instance.
+func New() *RootModule {
+	return &RootModule{
+		fileRegistry: fileRegistry{
+			files: make(map[string]*[]byte),
+		},
+	}
+}
+
+// NewModuleInstance implements the modules.Module interface and returns a new
+// instance of our module for the given VU.
+func (rm *RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	return &ModuleInstance{vu: vu, fileRegistry: &rm.fileRegistry}
+}
+
+// Exports implements the modules.Module interface and returns the exports of
+// our module.
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Named: map[string]any{
+			"open":     mi.Open,
+			"File":     file{},
+			"FileInfo": fileInfo{},
```

Comment:
We don't expect these to be directly usable in scripts, right? I.e. they're only exported in case the user wants to do some `instanceof` checks, or something like that?

Reply:
I don't expect users to use them directly indeed, but as you mentioned, it could be convinient to be able to `instanceof` etc on them. I'm open to removing though 👍🏻

Reply:
Sure, no worries, we can keep them. Was just wondering if you had other plans for them.